src/compiler.rs

```diff
@@ -75,14 +86,65 @@ impl Compiler {
               .join(Self::expand_tilde(&relative.cooked)?)
               .lexiclean();
 
-            if import.is_file() {
+            let has_glob = import.as_os_str().as_encoded_bytes().contains(&b'*');
+
+            if has_glob {
+              if !(unstable || ast_unstable) {
+                return Err(Error::UnstableFeature {
+                  unstable_feature: UnstableFeature::GlobImport,
+                });
+              }
+
+              let import_base_str =
+                import
+                  .as_os_str()
+                  .to_str()
+                  .ok_or_else(|| Error::ImportGlob {
+                    error: "non-UTF-8 glob import".to_string(),
+                    path: *path,
+                  })?;
+              let glob_options = glob::MatchOptions {
+                case_sensitive: true,
+                require_literal_separator: false,
+                require_literal_leading_dot: false,
+              };
+              let import_paths =
+                glob::glob_with(import_base_str, glob_options).map_err(|error| {
+                  Error::ImportGlob {
+                    error: error.to_string(),
+                    path: *path,
+                  }
+                })?;
+              for import in import_paths {
+                let Ok(import) = import else { continue };
+
+                if import.is_file() {
+                  if current.file_path.contains(&import) {
+                    return Err(Error::CircularImport {
+                      current: current.path,
+                      import,
+                    });
+                  }
+                  absolute_paths.push(import.clone());
+                  stack.push(current.import(import, path.offset));
+                } else if import.is_dir() {
+                  return Err(Error::ImportGlob {
+                    error: format!(
+                      "a glob import cannot match a directory; matched `{}`",
+                      import.display()
+                    ),
+                    path: *path,
+                  });
+                }
+              }
+            } else if import.is_file() {
```

Comment:
Is this circular import test duplicated with the one above?

Reply:
It is, but I think that trying to de-duplicate those lines of code would add more complexity than it's worth, since the logic here is just constructing the `CircularImport` error if necessary and then manipulating some local variables.